File: 002/solution.py
TARGET = 4e6
fibs = [1,1]
fib = 0
counter = 2
SUM = 0
while fib<TARGET:
    fib = fibs[counter-1]+fibs[counter-2]
    counter += 1
    fibs.append(fib)
    if fib%2==0:
        SUM += fib

print SUM

File: 002/solution.hs
{-stupid and straight-forward-}
fib 0 = 0
fib 1 = 1
fib n = fib (n-1) + fib (n-2)

{-http://blog.srinivasan.biz/software/fibonacci-numbers-the-slow-way-or-the-fast-and-lazy-way-}
{-fast-}
fibs = 1:1:zipWith (+) fibs (tail fibs)

every n xs = case drop (n-1) xs of
				(y:ys) -> y : every n ys
				[] -> []

main = do
		print $ sum evenfib
			where evenfib = takeWhile (<4000000) (every 3 fibs)

